base/applications/network/tracert/tracert.cpp

```diff
@@ -376,29 +376,30 @@ RunTraceRoute()
         return false;
     }
 
-    BYTE SendBuffer[PACKET_SIZE];
-    ICMPV6_ECHO_REPLY ReplyBufferv6;
+    struct SendAndReplyStruct
+    {
+        BYTE SendBuffer[PACKET_SIZE];
+        ICMPV6_ECHO_REPLY ReplyBufferv6;
 #ifdef _WIN64
-    ICMP_ECHO_REPLY32 ReplyBufferv432;
+        ICMP_ECHO_REPLY32 ReplyBufferv432;
 #else
-    ICMP_ECHO_REPLY ReplyBufferv4;
+        ICMP_ECHO_REPLY ReplyBufferv4;
 #endif
+    } SendAndReplyBuffer;
+
     PVOID ReplyBuffer;
 
-    DWORD ReplySize = PACKET_SIZE + SIZEOF_ICMP_ERROR + SIZEOF_IO_STATUS_BLOCK;
+    DWORD ReplySize = sizeof(SendAndReplyBuffer) + SIZEOF_ICMP_ERROR + SIZEOF_IO_STATUS_BLOCK;
     if (Info.Family == AF_INET6)
     {
-        ReplyBuffer = &ReplyBufferv6;
-        ReplySize += sizeof(ICMPV6_ECHO_REPLY);
+        ReplyBuffer = &(SendAndReplyBuffer.ReplyBufferv6);
     }
     else
     {
 #ifdef _WIN64
-        ReplyBuffer = &ReplyBufferv432;
-        ReplySize += sizeof(ICMP_ECHO_REPLY32);
+        ReplyBuffer = &(SendAndReplyBuffer.ReplyBufferv432);
 #else
-        ReplyBuffer = &ReplyBufferv4;
-        ReplySize += sizeof(ICMP_ECHO_REPLY);
+        ReplyBuffer = &(SendAndReplyBuffer.ReplyBufferv4);
```

Comment:
Unfortunately, this fixes nothing. `ReplyBuffer` and `ReplySize` must be consistent with regard to one another.
Looking at this case as an example, `ReplyBuffer` now points to a single `ICMP_ECHO_REPLY` structure, however `ReplySize` is now even larger than before your change.

The documentation for `Icmp6SendEcho2` gives us the reason why `ReplySize` must be the way it is (that is, the way it was before your change). The fix, therefore, is to make sure `ReplyBuffer` (wherever it may point) is guaranteed to have that much continuous free space -- either by placing it in a struct with the additional bytes available behind the buffer, or (perhaps simpler) by making it a heap allocation.